src/legendhpges/utils.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+import numpy as np
+
+
+def convert_coords(coords: np.ndarray) -> np.ndarray:
+    """Converts (x,y,zx) coordinates into (r,z)
+
+    Parameters
+    ----------
+    coords
+        numpy array of coordinates where the second index corresponds to (x,y,z) respectively
+
+    Returns
+    -------
+        numpy array of (r,z) coordinates for each point
+
+    """
+    r = np.sqrt(np.power(coords[:, 0], 2) + np.power(coords[:, 1], 2))
+    return np.column_stack((r, coords[:, 2]))
+
+
+def distance(a: np.ndarray, b: np.ndarray) -> np.ndarray:
+    """Computes the distance between a set of vectors a and b
+
+    Parameters
+    ----------
+    a
+        First list of vectors, where the second index corresponds to the dimension.
+    b
+        Second list in the same format.
+
+    Returns
+    -------
+        the distance for each vector
+    """
+    return np.sqrt(np.sum(np.power(a - b, 2), axis=1))
+
+
+def norm(a: np.ndarray) -> np.ndarray:
+    """Computes the norm of a set of vectors or a single vector
+
+    Parameters
+    ----------
+    a
+        First list of vectors, or a single vector.
+
+    Returns
+    -------
+        the length for each vector.
+    """
+    ax = 1 if a.ndim == 2 else 0
+
+    return np.sqrt(np.sum(np.power(a, 2), axis=ax))
+
+
+def dot(a: np.ndarray, b: np.ndarray) -> np.ndarray:
+    """Computes the dot product of a set of vectors
+
+    Parameters
+    ----------
+    a
+        First list of vectors, where the second index corresponds to the dimension.
+
+    b
+        Second list in the same format.
+
+
+    Returns
+    -------
+        the dot product for each vector.
+    """
+
+    return np.sum(a * b, axis=1)
+
+
+def shortest_distance_to_plane(
+    a_vec: np.array,
+    d: float,
+    points: np.ndarray,
+    rmax: float | None = None,
+    zrange: tuple[float, float] | None = None,
+) -> np.ndarray:
+    """Get the shortest distance from a plane (constrained in r and z) to each point.
+
+    The equation of the plane is given by :math:`a_1x+a_2y+a_3z=d`. Where :math:`\\vec{a}=(a_1,a_2,a_3)` The closest point on the plane to the point
```

Comment:
Wrap long lines in the docstrings, you can probably also set your text editor to do this automatically